### PR TITLE
Cockroach DB Dashboard Variables and enhancements.

### PR DESCRIPTION
**Pull request by gsidhwani-nr**
Added Dashboard Variables and enhancements.

# Summary

A concise description of the changes being introduced. Please review the pre-merge checklist section to validate this pull request is ready for review and merge. **If it is not ready, please mark the pull request as a draft.**

## Pre merge checklist

- [X ] Did you check you NRQL syntax? - Does it work?
- [X ] Did you include an InstallPlan and Documentation reference?
- [ X] Are all documentation links publically accessible?
- [ X] Did you check your descriptive content for [voice and tone](https://docs.newrelic.com/docs/style-guide/writing-strategies/voice-strategies-docs-sound-new-relic/)? 
- [ X] Did you check your descriptive content for spelling and grammar errors?

### Dashboards 

- [ X] Does the PR contain a screenshot for each of your dashboards?
- [X ] Do your screenshots show data?
- [ X] Has the [sanitization script](https://github.com/newrelic/newrelic-quickstarts/blob/main/CONTRIBUTING.md#dashboards) been run on each dashboard?

https://staging.onenr.io/0Zyw4XndLw3

### Alerts

- [X ] Did you check that your alerts actually work?

### CODE REVIEW

**Review by thezackm — APPROVED** [review]
approved for `prometheus-growth` project

File: quickstarts/cockroach-db/config.yml [hunk 1]
```diff
@@ -25,18 +27,21 @@ keywords:
   - k8s
   - prometheus
   - remote-write
+  - sql
 
 installPlans:
   - setup-prometheus
+  - setup-prometheus-agent-mode
```
Comment (thezackm) [inline quickstarts/cockroach-db/config.yml:34]: pending in https://github.com/newrelic/newrelic-quickstarts/pull/1503

File: dashboards/cockroach-db/cockroach-db.json [hunk 2]
```diff
@@ -68,7 +68,7 @@
...
+                "query": "SELECT  sum(`Max_capacity`) As 'Max Capacity', ((average(`Max_capacity`) * cardinality(`Max_capacity`)) - (average(`Available_capacity`) * cardinality(`Available_capacity`))) As 'Used Capacity' ,  sum(`Available_capacity`) As 'Avaliable Capacity' FROM (SELECT (average(capacity) * cardinality(capacity)) AS `Max_capacity`, (average(capacity_available) * cardinality(capacity_available)) AS `Available_capacity` FROM Metric  WHERE  (app_kubernetes_io_name LIKE '%cockroach%' or app_newrelic_com_name LIKE '%cockroach%' or k8s_app LIKE '%cockroach%') AND (instrumentation.name = 'remote-write') AND (prometheus_server IN ({{prometheus_server}})) AND (instance IN ({{instance}})) FACET ({{dimensions}}) ) FACET ({{dimensions}})"
```
Comment (thezackm) [inline dashboards/cockroach-db/cockroach-db.json:71]:
```
FACET ({{dimensions}}) ) FACET ({{dimensions}})
```

2 facets on this condition, please remove 1

File: dashboards/cockroach-db/cockroach-db.json [hunk 3]
```diff
@@ -95,7 +95,7 @@
...
-                "query": "SELECT (average(`sys_fd_open_number`) * cardinality(`sys_fd_open_number`)) As 'Open(Avg)', sum(sys_fd_softlimit_number) AS 'Soft Limit' FROM (SELECT (average(sys_fd_open) * cardinality(sys_fd_open)) AS `sys_fd_open_number`,  (average(sys_fd_softlimit) * cardinality(sys_fd_softlimit)) AS `sys_fd_softlimit_number` FROM Metric WHERE (app_kubernetes_io_name LIKE '%cockroach%' or app_newrelic_com_name LIKE '%cockroach%') And (instrumentation.name='remote-write') FACET instance ) FACET instance"
+                "query": "SELECT (average(`sys_fd_open_number`) * cardinality(`sys_fd_open_number`)) As 'Open(Avg)', sum(sys_fd_softlimit_number) AS 'Soft Limit' FROM (SELECT (average(sys_fd_open) * cardinality(sys_fd_open)) AS `sys_fd_open_number`,  (average(sys_fd_softlimit) * cardinality(sys_fd_softlimit)) AS `sys_fd_softlimit_number` FROM Metric WHERE  (app_kubernetes_io_name LIKE '%cockroach%' or app_newrelic_com_name LIKE '%cockroach%' or k8s_app LIKE '%cockroach%') AND (instrumentation.name = 'remote-write') AND (prometheus_server IN ({{prometheus_server}})) AND (instance IN ({{instance}})) FACET ({{dimensions}}) ) FACET ({{dimensions}})"
```
Comment (thezackm) [inline dashboards/cockroach-db/cockroach-db.json:98]:
```
FACET ({{dimensions}}) ) FACET ({{dimensions}})
```

2 facets on this condition, please remove 1

File: dashboards/cockroach-db/cockroach-db.json [hunk 4]
```diff
@@ -919,11 +939,11 @@
       "description": null,
       "widgets": [
         {
-          "title": "Active Nodes ",
+          "title": "Active Nodes ( Cluster)",
```
Comment (thezackm) [inline dashboards/cockroach-db/cockroach-db.json:942]:
```
Active Nodes ( Cluster)
```

typo: extra space in `(Cluster)`

File: dashboards/cockroach-db/cockroach-db.json [hunk 5]
```diff
@@ -946,44 +966,38 @@
...
               "ignoreTimeRange": false
-            },
-            "yAxisLeft": {
-              "zero": false
             }
           }
         },
         {
-          "title": "Garbage Collection Runs / Seconds ( All Nodes)",
+          "title": "Average Memory Usage ( selected servers and instances)",
```
Comment (thezackm) [inline dashboards/cockroach-db/cockroach-db.json:996]:
```
Average Memory Usage ( selected
```

typo: extra space `(selected`

File: dashboards/cockroach-db/cockroach-db.json [hunk 6]
```diff
@@ -1211,21 +1246,128 @@
...
+        {
+          "value": {
+            "string": "*"
+          }
+        }
+      ],
+      "nrqlQuery": {
+        "accountIds": [
+          0
+        ],
```
Comment (thezackm) [inline dashboards/cockroach-db/cockroach-db.json:1281]: set to `"accountIds": []`

File: dashboards/cockroach-db/cockroach-db.json [hunk 7]
```diff
@@ -1211,21 +1246,128 @@
...
```
Comment (thezackm) [inline dashboards/cockroach-db/cockroach-db.json:1302]: set to `"accountIds": []`

**Review by thezackm — CHANGES_REQUESTED** [review]
@gsidhwani-nr - changes requested; ty

File: dashboards/cockroach-db/cockroach-db.json (hunk 2 above)
Comment (gsidhwani-nr) [reply dashboards/cockroach-db/cockroach-db.json:71]: This is nested query and facets are part of respective queries and not for same queries. both are needed and same have been tested. 
Making no changes here.

File: dashboards/cockroach-db/cockroach-db.json (hunk 3 above)
Comment (gsidhwani-nr) [reply dashboards/cockroach-db/cockroach-db.json:98]: This is nested query and facets are part of respective queries and not for the same queries. Both facets are needed and same have been tested. 
Making no changes here.

File: dashboards/cockroach-db/cockroach-db.json (hunk 4 above)
Comment (gsidhwani-nr) [reply dashboards/cockroach-db/cockroach-db.json]: Modified.

File: dashboards/cockroach-db/cockroach-db.json (hunk 5 above)
Comment (gsidhwani-nr) [reply dashboards/cockroach-db/cockroach-db.json]: Modified.

File: dashboards/cockroach-db/cockroach-db.json (hunk 6 above)
Comment (gsidhwani-nr) [reply dashboards/cockroach-db/cockroach-db.json]: Modified

File: dashboards/cockroach-db/cockroach-db.json (hunk 7 above)
Comment (gsidhwani-nr) [reply dashboards/cockroach-db/cockroach-db.json]: Modified.

**Review by thezackm — APPROVED** [review]

**Review by josephgregoryii — APPROVED** [review]

**Pull request merged**